xtable-service/src/main/java/org/apache/xtable/service/ConversionService.java

```diff
@@ -0,0 +1,151 @@
+ */
+ 
+package org.apache.xtable.service;
+
+import static org.apache.xtable.model.storage.TableFormat.DELTA;
+import static org.apache.xtable.model.storage.TableFormat.HUDI;
+import static org.apache.xtable.model.storage.TableFormat.ICEBERG;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.conf.Configuration;
+
+import org.apache.hudi.common.table.timeline.HoodieInstant;
+
+import org.apache.iceberg.Snapshot;
+
+import org.apache.xtable.conversion.ConversionConfig;
+import org.apache.xtable.conversion.ConversionController;
+import org.apache.xtable.conversion.ConversionSourceProvider;
+import org.apache.xtable.conversion.SourceTable;
+import org.apache.xtable.conversion.TargetTable;
+import org.apache.xtable.delta.DeltaConversionSourceProvider;
+import org.apache.xtable.hudi.HudiConversionSourceProvider;
+import org.apache.xtable.iceberg.IcebergConversionSourceProvider;
+import org.apache.xtable.service.models.ConvertTableRequest;
+import org.apache.xtable.service.models.ConvertTableResponse;
+import org.apache.xtable.service.models.RestTargetTable;
+import org.apache.xtable.service.spark.SparkHolder;
+import org.apache.xtable.service.utils.DeltaMetadataUtil;
+import org.apache.xtable.service.utils.HudiMedataUtil;
+import org.apache.xtable.service.utils.IcebergMetadataUtil;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
+@ApplicationScoped
+public class ConversionService {
+  private final SparkHolder sparkHolder;
+  private final ConversionControllerFactory controllerFactory;
+  private final IcebergMetadataUtil icebergUtil;
+  private final HudiMedataUtil hudiUtil;
+  private final DeltaMetadataUtil deltaUtil;
+
+  @Inject
+  public ConversionService(
+      SparkHolder sparkHolder,
+      ConversionControllerFactory controllerFactory,
+      IcebergMetadataUtil icebergUtil,
+      HudiMedataUtil hudiUtil,
+      DeltaMetadataUtil deltaUtil) {
+    this.sparkHolder = sparkHolder;
+    this.controllerFactory = controllerFactory;
+    this.icebergUtil = icebergUtil;
+    this.hudiUtil = hudiUtil;
+    this.deltaUtil = deltaUtil;
+  }
+
+  public ConvertTableResponse convertTable(ConvertTableRequest request) {
+    Configuration conf = sparkHolder.jsc().hadoopConfiguration();
+    ConversionController conversionController = controllerFactory.create(conf);
+
+    SourceTable sourceTable =
+        SourceTable.builder()
+            .name(request.getSourceTableName())
+            .basePath(request.getSourceTablePath())
+            .formatName(request.getSourceFormat())
+            .build();
+
+    List<TargetTable> targetTables = new ArrayList<>();
+    for (String targetFormat : request.getTargetFormats()) {
+      TargetTable targetTable =
+          TargetTable.builder()
+              .name(request.getSourceTableName())
+              .basePath(request.getSourceTablePath())
+              .formatName(targetFormat)
```

Comment:
These should be target format and paths?

Reply:
So the client is only providing us the following currently, they techinally dont provide us with an explcit target format table name, or target format path. they only provide the target format enum (such as HUDI, ICEBERG, DELTA) 
```
{
  "source-format": "string",
  "source-table-name": "string",
  "source-table-path": "string",
  "target-formats": [
    "string"
  ],
  "configurations": {
    "additionalProp1": "string",
    "additionalProp2": "string",
    "additionalProp3": "string"
  }
}
```

In this case though my assumption is that the target table uses the same table name, and same table base path.

Reply:
Synced with Vinish the above assumption, we will be keeping this for now